pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentPushUtils.java

```diff
@@ -87,6 +89,7 @@ public static URI generateSegmentTarURI(URI dirURI, URI fileURI, String prefix,
   public static void pushSegments(SegmentGenerationJobSpec spec, PinotFS fileSystem, List<String> tarFilePaths)
       throws RetriableOperationException, AttemptsExceededException {
     String tableName = spec.getTableSpec().getTableName();
+    boolean cleanUpOutputDir = spec.isCleanUpOutputDir();
```

Comment:
Any specific reason to create this variable, as opposed to using `spec.isCleanupOutputDir()` inline, given that it is just used in one place?

Reply:
For readability and `spec.isCleanUpOutputDir()` is a kind of constant within the method which does not change in the nested for-loop thought defining a variable in outer scope will be helpful. Let me know if you would want to just go with  `spec.isCleanUpOutputDir()`.